model/riscv_insts_crypto.sail

```diff
@@ -0,0 +1,185 @@
+
+/*
+ * file: riscv_insts_crypto.sail
+ *
+ * This file contains instructions whos definitions do not change whether
+ * XLEN is 32 or 64.
+ *
+ * This file must be included in the model build whatever the value of XLEN.
+ *
+ */
+
+/*
+ * Scalar Cryptography Extension - Scalar        SHA256 instructions
+ * ----------------------------------------------------------------------
+ */
+
+/* AST clauses for the SHA256 instructions. */
+union   clause ast    =    SHA256SIG0 : (regidx,regidx)
+union   clause ast    =    SHA256SIG1 : (regidx,regidx)
+union   clause ast    =    SHA256SUM0 : (regidx,regidx)
+union   clause ast    =    SHA256SUM1 : (regidx,regidx)
+
+
+/* Encode/decode mappings for the SHA256 instructions. */
+mapping clause encdec =      SHA256SUM0   (rs1,rd) 
+    <-> 0b00 @ 0b01000 @ 0b00000 @ rs1 @ 0b001 @ rd @ 0b0010011
+
+mapping clause encdec =      SHA256SUM1   (rs1,rd) 
+    <-> 0b00 @ 0b01000 @ 0b00001 @ rs1 @ 0b001 @ rd @ 0b0010011
+
+mapping clause encdec =      SHA256SIG0   (rs1,rd) 
+    <-> 0b00 @ 0b01000 @ 0b00010 @ rs1 @ 0b001 @ rd @ 0b0010011
+
+mapping clause encdec =      SHA256SIG1   (rs1,rd) 
+    <-> 0b00 @ 0b01000 @ 0b00011 @ rs1 @ 0b001 @ rd @ 0b0010011
+
+
+/* Assembly string mappings for the SHA256 instructions. */
+mapping clause assembly = SHA256SIG0 (rs1,rd) <-> "sha256sig0" ^ spc() ^ reg_name(rd) ^ sep() ^ reg_name(rs1)
+mapping clause assembly = SHA256SIG1 (rs1,rd) <-> "sha256sig1" ^ spc() ^ reg_name(rd) ^ sep() ^ reg_name(rs1)
+mapping clause assembly = SHA256SUM0 (rs1,rd) <-> "sha256sum0" ^ spc() ^ reg_name(rd) ^ sep() ^ reg_name(rs1)
+mapping clause assembly = SHA256SUM1 (rs1,rd) <-> "sha256sum1" ^ spc() ^ reg_name(rd) ^ sep() ^ reg_name(rs1)
+
+val      crypto_sha256 : (sha256_op, regidx, regidx) -> Retired effect {escape,rreg,wreg}
+/* --- specification snippet begin --- */
+function crypto_sha256   (op        , rd    , rs1   ) =  {
+  let inb    : bits(32) = X(rs1)[31..0];
+  let result : bits(32) = match op {
+    OP_SHA256_SIG0 => ror32(inb ,07) ^ ror32(inb ,18) ^ (inb >>  3),
```

Comment:
Commas in the wrong places

Reply:
For the avoidance of doubt, the following is what you're after:
```
function crypto_sha256   (op, rd, rs1) =  { 
[...]
    OP_SHA256_SIG0 => ror32(inb, 07) ^ ror32(inb, 18) ^ (inb >>  3),
```
right?

Reply:
For this, and the other "commas" comments, yes?

Reply:
The final line was specifically what I was commenting on (though I note now you have two spaces after the `>>` and before the `{`). I didn't care so much about the function parameters, though it isn't really the current style and probably would be better done more normally like your comment.